src/machine/machine_nrf52840.go

```diff
@@ -29,3 +29,20 @@ func (i2c I2C) setPins(scl, sda uint8) {
 	i2c.Bus.PSEL.SCL = nrf.RegValue(scl)
 	i2c.Bus.PSEL.SDA = nrf.RegValue(sda)
 }
+
+// SPI
+func (spi SPI) setPins(sck, mosi, miso uint8) {
+	var s, mo, mi = sck, mosi, miso
+	if s == 0 {
+		s = SPI0_SCK_PIN
```

Comment:
This idea was just to make sure that there is a sensible default if you do not enter a value. This will also accommodate the signature when you cannot change the pins in use, for example AVR.